starlite/middleware/session/base.py

```diff
@@ -0,0 +1,313 @@
+import abc
+import secrets
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Awaitable,
+    Callable,
+    Dict,
+    Generic,
+    Optional,
+    Type,
+    TypeVar,
+    Union,
+    cast,
+)
+
+from orjson import OPT_SERIALIZE_NUMPY, dumps, loads
+from pydantic import BaseConfig, BaseModel, PrivateAttr, conint, conlist, constr
+from starlette.datastructures import MutableHeaders
+from typing_extensions import Literal
+
+from starlite import ASGIConnection, Cookie, DefineMiddleware
+from starlite.middleware.base import MiddlewareProtocol
+from starlite.utils import default_serializer, get_serializer_from_scope
+
+if TYPE_CHECKING:
+    from starlite.types import ASGIApp, Message, Receive, Scope, ScopeSession, Send
+
+
+ONE_DAY_IN_SECONDS = 60 * 60 * 24
+
+
+class BaseBackendConfig(BaseModel):
+    _backend_class: Type["SessionBackend"] = PrivateAttr()
```

Comment:
So, this is repeating everywhere - what I suggest is to modify this as such:

```python
TBackend = TypeVar("TBackend", bound=Type["SessionBackend"])

class BaseBackendConfig(GenericModel, Generic[TBackend]):
   backend_class: TBackend = PrivateAttr()
```

You can then use it using a generic.

Reply:
That's what I had initially, but it broke mypy. I was hitting some version of [this](https://github.com/python/mypy/issues/12113). I wanted to revisit this at a later point and see if I can either circumnavigate this somehow, or otherwise file a bug with mypy.

Reply:
Yhea, this is the same stuff I got it in ResponseContainers. No clue wtf.

Reply:
It's really annoying because having this properly typed all the way through would make some things easier.

I had something like this on the base config class:

```python
@property
def backend(self) -> "BackendT"
    return self._backend_class(config=self)
```

But this is only really useful if it type checks correctly and doesn't just return the abstract classes type.